rust/cargo-psibase/src/main.rs

```diff
@@ -325,108 +350,178 @@ async fn build(
     Ok(files)
 }
 
-async fn get_test_services(args: &Args) -> Result<Vec<(String, String)>, Error> {
-    let mut command: tokio::process::Child = tokio::process::Command::new(get_cargo())
-        .arg("test")
+async fn build_plugin(
+    args: &Args,
+    packages: &[&str],
+    extra_args: &[&str],
+) -> Result<Vec<PathBuf>, Error> {
+    let mut command = tokio::process::Command::new(get_cargo())
+        .arg("component")
```

Comment:
Does that do the right thing? We want the `cargo-psibase` binary to require a `cargo-component` binary. We don't want the `cargo-psibase` binary to depend `cargo-component` built as a lib.

Reply:
Well, we actually could depend on it as a lib and then call the cargo-component command directly, which would have helpful side effects:
1. Reducing the number of steps needed in a from-scratch environment setup
2. Pinning the specified version of cargo-component

Reply:
I'm realizing just now that I am pretty sure `cargo component build --release` is the same as, `cargo build --release --target=wasm32-wasi`

Reply:
Doesn't `cargo component` need to update `bindings.rs` and also do some post-processing?

Reply:
I'm somewhat hesitant to use `cargo-component` as a library, because I don't think it's really intended to be used that way. cargo itself says that its library interface is unstable and it should be called on the command line.

Reply:
> Doesn't `cargo component` need to update `bindings.rs` and also do some post-processing?

Hmm, that would make sense. I guess `build` is forwarded to cargo ([Source](https://github.com/bytecodealliance/cargo-component?tab=readme-ov-file#usage)), but maybe cargo component does these other things pre/post `cargo build`.